src/core/tests/type_prop/gather.cpp

```diff
@@ -723,38 +684,50 @@ TEST(type_prop, gather_8_dynamic_value_and_label_propagation) {
     ASSERT_EQ(ov::DimensionTracker::get_label(output_shape[0]), 10);
 }
 
+/** \brief Check usage of evaluate lower/upper and label on shape inference. */
+TEST(type_prop, gather_v8_dynamic_value_and_label_propagation_interval_dim) {
+    Dimension marked_0 = Dimension(2, 4);
+    ov::DimensionTracker::set_label(marked_0, 10);
+    PartialShape target_0 = PartialShape{marked_0, 4};
+
+    auto param = std::make_shared<op::Parameter>(element::f32, Shape{1});
+    auto param_0 = std::make_shared<op::Parameter>(element::f32, target_0);
+    auto shape_0 = std::make_shared<op::ShapeOf>(param_0);
+
+    const auto& et = element::i64;
+    std::vector<int64_t> zero{0};
+    const auto indices = std::make_shared<op::v0::Constant>(et, Shape{zero.size()}, zero);
+    const auto axis = std::make_shared<op::v0::Constant>(et, Shape{}, zero);
+    const auto gather = std::make_shared<op::v8::Gather>(shape_0, indices, axis);
+
+    auto bc = std::make_shared<op::v1::Broadcast>(param, gather);
+    ASSERT_EQ(bc->get_output_partial_shape(0), PartialShape({marked_0}));
```

Comment:
why do we use assert? I suppose we need to use except. The difference here that assert aborts execution, while expect allows to proceed with next checks.

Reply:
Just follow what is used in this file, but true `expect` will be better. Will update all checks in this test file.